File: eeducation/src/main/kotlin/rs/eeducation/service/SearchService.kt
package rs.eeducation.service

import org.springframework.stereotype.Service
import rs.eeducation.model.Course
import rs.eeducation.model.School

@Service
class SearchService(private val courseService: CourseService, private val schoolService: SchoolService) {

    fun searchSchools(name: String): List<School> {
        return schoolService.findSchoolsByName(name)
    }

    fun searchCourses(name: String): List<Course> {
        return courseService.findByName(name)
    }
}

File: eeducation-front/src/app/service/search.service.ts
import {Injectable} from '@angular/core';
import {PathService} from './path.service';
import {HttpClient} from '@angular/common/http';
import {CourseBasic} from '../model/CourseBasic';
import {Observable} from 'rxjs';
import {School} from '../model/School';

@Injectable({
  providedIn: 'root'
})
export class SearchService {

  private path = '';

  constructor(private pathService: PathService, private http: HttpClient) {
    this.path = pathService.path + '/api/search';
  }

  searchCourses(name: string): Observable<Array<CourseBasic>> {
    return this.http.get<Array<CourseBasic>>(this.path + '/course/' + name);
  }

  searchSchools(name: string): Observable<Array<School>> {
    return this.http.get<Array<School>>(this.path + '/school/' + name);
  }
}
